src/FedoraApi.php

```diff
@@ -305,4 +307,79 @@ public function getGraph(ResponseInterface $response)
         }
         return $graph;
     }
+
+    /**
+     * Creates version in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param string $timestamp Timestamp for Memento version
+     * @param string $content String or binary content
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function createVersion(
+        $uri = '',
+        $timestamp = '',
+        $content = null,
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false];
+        if ($timestamp != ''){
+            $headers['Memento-Datetime'] = $timestamp;
+        }
+        if ($content != null){
+            $options['body'] = $content;
+        }
```

Comment:
Because you must provide a body if you provide a timestamp (and vice versa) you probably need some sort of checking and return a (say) `400 Bad Request` if you only provide one without the other?

Reply:
That makes sense to me, just not sure how to return a 400 Bad Request. All of the other responses being returned in chullo are being returned from the guzzle client->request so would i need to mock somekind of 400 object to return instead?

Reply:
Hmmm you know, we should test that. If Fedora returns a 400 you might not have to do anything and this would pass that 400 response along to the caller.

Reply:
`'http_errors' => false` is being set, so this'll just return a 400+ response instead of throwing an exception :tada:

Reply:
👍 modified in 39bedbc